src/ghost.js

```diff
@@ -68,7 +68,16 @@ class GhostThread extends EventEmitter {
    * @return    {Array<Object>}      users online
    */
   async getPosts () {
-    return [...this._backlog].map(msg => JSON.parse(msg)).sort((p1, p2) => p1.timestamp - p2.timestamp)
+    const _12hrInSecs = 12 * 60 * 60
+    const posts = [...this._backlog]
+      .map(msg => JSON.parse(msg))
+      .sort((p1, p2) => p1.timestamp - p2.timestamp)
+      .filter(msg => msg.timestamp > (Math.floor(Date.now() / 1000) - _12hrInSecs))
+
+    posts.forEach(msg => {
+      this._backlog.add(JSON.stringify(msg))
```

Comment:
Not sure why you are doing this? It seems like these *msg* would already be in the `this._backlog` set?

Reply:
The idea was that if we keep a shorter version of the backlog it would make looping through it more efficient. Once we get a limited version of it we should just keep it on hand instead of always looping through the whole history.

Reply:
Ok, but then you need to clear out the `this._backlog` variable before right? Otherwise it would still contain all messages!